domain-check.bash

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+# This is meant to be an interim script to (1) evaluate we have
+# enough capabilities inside of the urlgetter pseudo-test and (2)
+# prototype the next generation Web Connectivity.
+#
+# Very alpha. Very basic. Needs much more love.
+#
+# Inspired by great chats and great code. Main source of ispiration is
+# currently https://github.com/Jigsaw-Code/net-analysis/blob/master/netanalysis/blocktest/measure.sh
+
+function fatal() {
+  echo "FATAL: $@" 1>&2
+  exit 1
+}
+
+if [ $# -ne 1 ]; then
+  fatal "usage: $0 domain"
+fi
+domain=$1
+
+function require() {
+  if ! [ -x "$(command -v $1)" ]; then
+    fatal "$1 not installed, please run: $2"
+  fi
+}
+
+require jq "sudo apt install jq"
+require uuidgen "sudo apt install uuid-runtime"
+
+uuid=$(uuidgen)
+
+function log() {
+  echo "$@" 1>&2
+}
+
+function run() {
+  log "$@"
+  "$@"
+}
+
+function urlgetter() {
+  #run ./miniooni -A session=$uuid $@ urlgetter
+  run ./miniooni -n --no-bouncer -A session=$uuid "$@" urlgetter
+}
+
+function getfailure() {
+  tail -n1 report.jsonl|jq -r .test_keys.failure
+}
+
+function output() {
+  echo "$@"
+}
+
+function getipv4first() {
+  tail -n1 report.jsonl|jq -r ".test_keys.queries|.[]|select(.hostname==\"$1\")|select(.query_type==\"A\")|.answers|.[0].ipv4"
+}
+
+function getipv4list() {
+  tail -n1 report.jsonl|jq -r ".test_keys.queries|.[]|select(.hostname==\"$1\")|select(.query_type==\"A\")|.answers|.[].ipv4"|sort
+}
+
+function getbody() {
+  # Implementation note: requests stored in LIFO order
+  tail -n1 report.jsonl|jq -r ".test_keys.requests|.[0]|.response.body"
+}
+
+testdomain=${MINIOONI_TEST_DOMAIN:-example.org}
```

Comment:
You may want to organize the code below in functions, so it's easier to read.

Reply:
Good point